lib/3scale/backend/stats/keys/generator/timestamp.rb

```diff
@@ -0,0 +1,126 @@
+module ThreeScale
+  module Backend
+    module Stats
+      module Keys
+        module Generator
+          class Timestamp
+            private
+            def get_metric_type_granularity_limits(periods)
+              # Array range goes from A to B, (B - A + 1) elements
+              return Range.new(0, periods.size - 1) if index_limits.nil?
+              limit_start = index_limits[0].granularity || 0
+              limit_end = index_limits[1].granularity || 0
+              Range.new(limit_start, limit_end)
+            end
+
+            def get_granularity_limits
+              [index_limits[0].granularity, index_limits[1].granularity]
+            end
+
+            def get_time_limits
+              Range.new(index_limits[0].ts, index_limits[1].ts)
```

Comment:
Range with timestamps???

Reply:
I'm not sure too of using Ranges in the case of limits/timestamps. Ranges were used at first as recommended by @unleashed and to keep consistency with all other kind of limits this was used here too. However, I'm still evaluating what would be the best way to do it as I keep doing refactors in some parts of the code